mycore-base/src/main/java/org/mycore/access/strategies/MCRCreatorRuleStrategy.java

```diff
@@ -118,24 +147,24 @@ public boolean hasRuleMapping(String id, String permission) {
     }
 
     public boolean isCreatorRuleAvailable(String id, String permission) {
-        if (CREATOR_PERMISSIONS.contains(permission)) {
-            MCRObjectID mcrObjectId = null;
-            try {
-                mcrObjectId = MCRObjectID.getInstance(id);
-                MCRUserInformation currentUser = MCRSessionMgr.getCurrentSession().getUserInformation();
-                if (CREATOR_ROLES.stream().anyMatch(currentUser::isUserInRole)
-                    && objectStatusIsSubmitted(mcrObjectId)) {
-                    if (isCurrentUserCreator(mcrObjectId, currentUser)) {
-                        return true;
-                    }
+        MCRObjectID mcrObjectId = null;
+        try {
+            mcrObjectId = MCRObjectID.getInstance(id);
+            MCRUserInformation currentUser = MCRSessionMgr.getCurrentSession().getUserInformation();
+            if (userHasCreatorRole(currentUser) && userIsCreatorOf(currentUser, mcrObjectId)) {
+                if (objectStatusIsSubmitted(mcrObjectId) && CREATOR_SUBMITTED_PERMISSIONS.contains(permission)) {
+                    return true;
                 }
-            } catch (RuntimeException e) {
-                if (mcrObjectId == null) {
-                    LOGGER.debug("id is not a valid object ID", e);
-                } else {
-                    LOGGER.warn("Eror while checking permission.", e);
+                if (objectStatusIsReview(mcrObjectId) && CREATOR_REVIEW_PERMISSIONS.contains(permission)) {
+                    return true;
                 }
             }
+        } catch (RuntimeException e) {
```

Comment:
The entire catch block is bad, but it was there before.

- Catching RuntimeException instead of a specific exception
- the mcrObjectId == null can be checked before

Reply:
There is no way for `mcrObjectId` to be `null` anyway. 

My reasoning: `MCRObjectID#getInstance(String)` will throw an exception if the provided string is `null`, Otherwise it will call `MCRObjectIDPool.getMCRObjectID` which will return a cached `MCRObjectID` or try to construct one. Construction will fail, if the provided string is not a syntactically valid object ID.

I'll remove the superfluous `if`-`then`-`else` but keep the `try`-`catch`. This PR should only extend the configuration options for the creator rule strategy, it shouldn't change the externally observed behavior  of this class. A follow-up PR might be appropriate.